include/lsst/meas/extensions/shapeHSM/HsmMomentsControl.h

```diff
@@ -158,25 +196,81 @@ class HsmSourceMomentsAlgorithm : public HsmMomentsAlgorithm {
     Control _ctrl;
 };
 
+
 /// Class to measure HSM adaptive moments of PSF
 class HsmPsfMomentsAlgorithm : public HsmMomentsAlgorithm {
 public:
     /// A typedef to the Control object for this algorithm, defined above.
     ///  The control object contains the configuration parameters for this algorithm.
-    typedef HsmPsfMomentsControl Control;
+    typedef std::shared_ptr<const HsmPsfMomentsControl> Control;
```

Comment:
This wasn't for the argument passing (or anything pybind11), but to enable one class hierarchy to hold onto instances of a different class hierarchy.  I.e., I needed a pointer instead of a concrete object so that `HsmPsfMomentsAlgorithm` could hold a `HsmPsfMomentsControl` and `HsmPsfMomentsDebiasedAlgorithm` could hold a `HsmPsfMomentsDebiasedControl`, and the debiased methods that call the parent methods still work.  If there's a different pattern you recommend, definitely let me know!

Reply:
Makes sense, and I don't have a better idea :+1:.